Feature/fe signup login pages

mehdikaray opened the pull request:
- add login and register pages
- remove ping and loading pages
- add global theme attribute 
- import google open sans font
- add app and login register layouts
- add react hook form for form validation 
- add login and register header 
- add route container to manage routes

![image](https://user-images.githubusercontent.com/66958060/90644346-866dc280-e22c-11ea-878a-6f5d7fc26742.png)

![image](https://user-images.githubusercontent.com/66958060/90644881-304d4f00-e22d-11ea-8027-f97a9f97b97a.png)

![image](https://user-images.githubusercontent.com/66958060/90645163-891ce780-e22d-11ea-9046-28f469e6c890.png)

mehdikaray committed:
ADD: add login and register pages

- add login and register pages
- remove ping and loading pages
- add global theme attribute 
- import google open sans font

mehdikaray committed:
ADD: add account and app header component

mehdikaray committed:
ADD: add app and account layout

mehdikaray committed:
ADD: add route container and routing system

mehdikaray committed:
ADD: add responsive  to login and register pages

skassam21 commented on client/src/App.css:
```diff
@@ -1,3 +1,5 @@
+@import url('https://fonts.googleapis.com/css2?family=Open+Sans&display=swap');
```
We might want to edit this in the public/index.html page and then we can also remove the existing font input that is not used

skassam21 commented on client/src/themes/theme.js:
```diff
@@ -2,13 +2,20 @@ import { createMuiTheme } from "@material-ui/core";
     h1: {
-      // could customize the h1 variant as well
-    }
+      fontSize: "40px",
```
For font sizes, let's not use pixels, instead you can just write fontSize: 40 (see https://material-ui.com/customization/typography/#font-size)

skassam21 reviewed (commented):
This looks great, just a few small comments, can you post some screenshots in the description for PRs that there are UI elements (only if it makes sense)

mehdikaray committed:
EDIT: fix code comments after code review 

- edit importing font from index.html
- edit font size syntax
- fix react hook form register ref in login page

skassam21 reviewed (approved):
Looks great 👍
